a10_octavia/controller/worker/flows/a10_member_flows.py

```diff
@@ -779,3 +779,240 @@ def get_create_member_snat_pool_subflow(self):
+                    id=m.id, flow=constants.MARK_MEMBER_ACTIVE_INDB)))
+
+        # for updating of members
+        batch_update_members_flow.add(
+            lifecycle_tasks.MembersToErrorOnRevertTask(
+                name='{flow}-updated'.format(
+                    flow=constants.MEMBER_TO_ERROR_ON_REVERT_FLOW),
+                # updated_members is a list of (obj, dict), only pass `obj`
+                inject={constants.MEMBERS: [m[0] for m in updated_members]}))
+        for m, um in updated_members:
+            um.pop('id', None)
+            batch_update_members_flow.add(database_tasks.MarkMemberPendingUpdateInDB(
+                inject={constants.MEMBER: m},
+                name='mark-member-pending-update-in-db-' + m.id))
+            batch_update_members_flow.add(server_tasks.MemberUpdate(
+                name='member-update-' + m.id,
+                inject={constants.MEMBER: m},
+                requires=(constants.MEMBER, a10constants.VTHUNDER,
+                          constants.POOL, constants.FLAVOR)))
+            batch_update_members_flow.add(database_tasks.UpdateMemberInDB(
+                name='update-member-in-db-' + m.id,
+                inject={constants.MEMBER: m, constants.UPDATE_DICT: um}))
+            batch_update_members_flow.add(database_tasks.MarkMemberActiveInDB(
+                inject={constants.MEMBER: m},
+                name='{flow}-{id}'.format(
+                    id=m.id, flow=constants.MARK_MEMBER_ACTIVE_INDB)))
+
+        existing_members = [m[0] for m in updated_members]
+        pool_members = new_members + existing_members
+        if pool_members:
+            batch_update_members_flow.add(
+                self.get_handle_member_vrid_internal_subflow(pool_members))
+            if CONF.a10_global.network_type == 'vlan':
+                batch_update_members_flow.add(vthunder_tasks.TagInterfaceForMember(
+                    name='update_tagged_interface_for_members',
+                    inject={constants.MEMBER: pool_members},
+                    requires=[constants.MEMBER,
+                              a10constants.VTHUNDER]))
+
+        batch_update_members_flow.add(database_tasks.MarkPoolActiveInDB(
+            requires=constants.POOL))
+        batch_update_members_flow.add(database_tasks.MarkLBAndListenersActiveInDB(
+            requires=(constants.LOADBALANCER,
+                      constants.LISTENERS)))
+        batch_update_members_flow.add(vthunder_tasks.WriteMemory(
+            requires=a10constants.VTHUNDER))
+        batch_update_members_flow.add(a10_database_tasks.SetThunderUpdatedAt(
+            requires=a10constants.VTHUNDER))
+
+        return batch_update_members_flow
+
+    def get_batch_update_member_snat_pool_subflow(self, member):
+        batch_update_member_snat_subflow = linear_flow.Flow(
+            a10constants.CREATE_MEMBER_SNAT_POOL_SUBFLOW)
+        batch_update_member_snat_subflow.add(server_tasks.MemberFindNatPool(
```

Comment:
Can we just get all that natpools once in one call instead of making a new call per member?

Reply:
Done